sycl/include/CL/sycl/detail/aligned_allocator.hpp

```diff
@@ -53,6 +53,8 @@ template <typename T> class aligned_allocator {
   pointer allocate(size_t Size) {
     size_t NumBytes = Size * sizeof(value_type);
     NumBytes = ((NumBytes - 1) | (MAlignment - 1)) + 1;
+    if (0 == NumBytes)
+      throw std::bad_alloc();
```

Comment:
I am puzzled by what you are trying to test and to forbid...
It might be obvious since there is no comment. :-)
For example allocate 1 `char` with `MAlignment` = 0 would throw but 0 `char` with `MAlignment` = 1 is fine.

Reply:
@jbrodman, could you add a regression LIT test, please?
It should help with clarifying the case you fix and avoid regressions in the future.

Reply:
Done.